tpi_rs/src/middleware/persistency/app_persistency.rs

```diff
@@ -0,0 +1,282 @@
+use std::future;
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+
+use super::binary_persistency::PersistencyStore;
+use anyhow::Context;
+use futures::future::Either;
+use tokio::fs::{File, OpenOptions};
+use tokio::time::sleep_until;
+
+const BIN_DATA: &str = "/var/lib/bmcd/bmcd.bin";
+const WRITE_BACK_TIMEOUT: Duration = Duration::from_secs(300);
+
+#[derive(Debug)]
+enum MonitorEvent {
+    StoreChange,
+    PersistencyWritten,
+}
+
+/// Builder to aid in configuring and setting up a [ApplicationPersistency], a
+/// "key/value" store.
+#[derive(Debug)]
+pub struct PersistencyBuilder {
+    keys: Vec<(&'static str, Vec<u8>)>,
+    write_timeout: Option<Duration>,
+}
+
+impl PersistencyBuilder {
+    /// Add a key to the key/value store. Attempting to access keys that are not
+    /// registered with this function will result in an error.
+    pub fn register_key<T>(mut self, key: &'static str, default: T) -> Self
+    where
+        T: Send + serde::Serialize,
+    {
+        self.keys.push((key, bincode::serialize(&default).unwrap()));
+        self
+    }
+
+    /// The [ApplicationPersistency] contains a write mechanism that writes the
+    /// key/value store back to the file-system. This happens when on a timeout
+    /// occurrence started from the last write. This function disables the write
+    /// on timeout. The key/value store only gets written when the
+    /// [ApplicationPersistency] is dropped.
+    pub fn disable_write_on_timeout(mut self) -> Self {
```

Comment:
Doesn't seem to be used. Was this a debugging feature?

Reply:
It was a debug feature, but I also anticipate we need to have this at hand. Its a self documenting way to disable timeout writebacks when needed